### PR TITLE
Possibility to set persistent connection flag when creating socket

### PR DESCRIPTION
**Pull request by tenerd**
Added posibility to create socket with STREAM_CLIENT_PERSISTENT.

Can you point me what (unit) tests to create?

### CODE REVIEW

File: src/Stomp/Network/Connection.php [hunk 1]
```diff
@@ -110,14 +117,16 @@ class Connection
      *
      * @param string $brokerUri
      * @param integer $connectionTimeout in seconds
-     * @param array   $context stream context
+     * @param array $context stream context
+     * @param bool $persistentConnection
      * @throws ConnectionException
      */
-    public function __construct($brokerUri, $connectionTimeout = 1, array $context = [])
+    public function __construct($brokerUri, $connectionTimeout = 1, array $context = [], $persistentConnection = false)
```
Comment (staabm) [inline src/Stomp/Network/Connection.php]: instead of adding even more __construct() args I would prefer a dedicated setter.

Comment (jmglsn) [inline src/Stomp/Network/Connection.php]: Maybe we should add a setter for `stream_socket_client` - `flags`?

Comment (staabm) [inline src/Stomp/Network/Connection.php]: I see no more flags which would be usefull.
the benefit of having dedicated setters is, that the Connection class api hides the underlying implementation details.

File: src/Stomp/Network/Connection.php [hunk 2]
```diff
@@ -245,12 +254,16 @@ protected function connectSocket(array $host)
         $errNo = null;
         $errStr = null;
         $context = stream_context_create($this->context);
+        $flags = STREAM_CLIENT_CONNECT;
+        if ($this->persistentConnection) {
+            $flags = STREAM_CLIENT_CONNECT | STREAM_CLIENT_PERSISTENT;
```
Comment (OndraM) [inline src/Stomp/Network/Connection.php]: You can use just `$flags |= STREAM_CLIENT_PERSISTENT;`

File: src/Stomp/Network/Connection.php (hunk 1 above)
Comment (tenerd) [inline src/Stomp/Network/Connection.php]: Ok, constructor param replaced with setter. Does it look like you thought?:)

File: src/Stomp/Network/Connection.php (hunk 2 above)
Comment (tenerd) [inline src/Stomp/Network/Connection.php]: Thx

**Comment by staabm** [comment]
LGTM thx 👍

**Comment by tenerd** [comment]
I wanted to squash commits to one, but nevermind:)

**Comment by jmglsn** [comment]
Sorry too fast and maybe a little bit too early this day for me ;)

**Comment by jmglsn** [comment]
Anyway now it's available in https://github.com/stomp-php/stomp-php/releases/tag/4.1.3 thanks!

**Pull request merged**